mongoimport/mongoimport.go

```diff
@@ -502,8 +522,10 @@ func (up *upserter) Insert(doc interface{}) error {
 	var err error
 	if selector == nil {
 		err = up.collection.Insert(document)
-	} else {
+	} else if up.imp.IngestOptions.Mode == MODE_UPSERT {
 		_, err = up.collection.Upsert(selector, document)
+	} else { // mode == 'merge'
+		_, err = up.collection.Upsert(selector, bson.M{"$set": document})
```

Comment:
if I change the source to 
```go
_, err = up.collection.Upsert(selector, bson.D{{"$set": document}})
```

compilication error is
```sh
./mongoimport.go:530: invalid field name "$set" in struct initializer
```

I don't quite understand. bson.M to wrap {$set:...} is the only way I found working right now